### PR TITLE
[Themes] Add new semantic color scheme

### PR DESCRIPTION
**Pull request by romoore**
The new semantic color scheme includes a number of new properties focused on their contextual use within components. This should make it easier to provide consistent, holistic theming for an entire app using a set of baseline colors and mappings.

## "Color Scheme" Schema
* `primaryColor` - Displayed most frequently across your app.
* `primaryColorVariant`  - A tonal variation of primary color.
* `secondaryColor` - Accents select parts of your UI.
* `backgroundColor` - The underlying color of an app’s content.
* `errorColor` - The color used to indicate error status.
* `surfaceColor` - The color of surfaces such as cards, sheets, menus.
* `onBackgroundColor` - A color that passes accessibility guidelines for text/iconography when drawn on top of `backgroundColor`.
* `onPrimaryColor` - A color that passes accessibility guidelines for text/iconography when drawn on top of `primaryColor`.
* `onSecondaryColor` - A color that passes accessibility guidelines for text/iconography when drawn on top of `secondaryColor`.
* `onSurfaceColor` - A color that passes accessibility guidelines for text/iconography when drawn on top of `surfaceColor`.

**New Catalog Component**
screen shot 2018-03-30 at 3 20 23 pm

### CODE REVIEW

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (randallli) [inline components/Themes/src/MDCSemanticColorScheme.h]: Protocol docs.
We need to explain the difference between this and the MDCColorScheme protocol.

**Review by randallli — APPROVED** [review]

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: Done

**Review by randallli — DISMISSED** [review]

**Comment by romoore** [comment]
# Important!

Please don't merge until all reviewers have approved.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/Themes/src/MDCSemanticColorScheme.h]: We have to decide where to implement the fact that `secondaryColor` falls back to `primaryColor` if null. A couple options I see available to us:

- [ ] We make this property `nonnull` on the protocol and `null_resettable` on the concrete class. We document that setting `secondaryColor` to nil will result in `secondaryColor` returning `primaryColor`. This means components blindly pull from `secondaryColor` without implementing the fallback logic.
- [ ] We keep the current approach and implement the fallback logic in each component. This means each component's code is more explicit about the behavior of choosing either secondary or primary.

What do you think?

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/Themes/src/MDCSemanticColorScheme.h]: > If null, it is expected that the primaryColor will be used instead.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/Themes/src/MDCSemanticColorScheme.h]: Expose all of the properties on this class as writeable as well. It should be easy for someone to change the properties after initialization if they so desire (e.g. someone creates a Baseline color scheme initialized with the purples and then changes only the secondary color).

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: Another option:

- [ ] We make this property `nonnull` on the protocol and `nonnull` on the concrete class. We document in the concrete class that initializing with `secondaryColor` as nil will result in `secondaryColor` being assigned to `primaryColor`.

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: This is assuming the API is still nullable, pending the decision below.

File: components/Themes/src/MDCSemanticColorScheme.m
Comment (ianegordon) [inline components/Themes/src/MDCSemanticColorScheme.m]: Super-nit: CGFloat here and below.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/Themes/src/MDCSemanticColorScheme.h]: Attaching light and dark to these variants will make it harder to implement darkMode.
If lightMode, statusBar.background = primaryColorLightVariant
else if darkMode, statusBar.background = primaryColorDarkVariant

statusBar.background = primaryColorEmphasisVariant

Naming is hard!

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (ianegordon) [reply components/Themes/src/MDCSemanticColorScheme.h]: I vote for option two.
initWithPrimary:(UIColor) // Set all the properties, including calculating a sane secondary
initWithPrimary:secondary:  // Set all the properties, based on the params

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (ianegordon) [reply components/Themes/src/MDCSemanticColorScheme.h]: Consider:
initWithPrimary:  // Calculate a sane secondary, error, etc.

I haven't seen this method of using nullable parameters as optionals before, typically I would expect all the parameters to be nonnull and there to be convenience constructors all wired into the designated init.

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: My understanding presently is that these properties are semantically safe as light/dark variants regardless of the theme being light or dark, but I also have not seen any literal mappings be well defined in any componentry, so the jury's out on whether that's actually the case...

Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: At least for now, this is what our cross-platform spec is expecting as far as symbols.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: We are intentionally not doing any code-side calculation of colors in the short term. That logic will live solely in the color tool.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: Should the "secondary falls back to primary" be part of the protocol or not? Right now it's only in the concrete implementation.

@ianegordon I agree with @jverkoey that the new theme should not do any internal calculations that aren't explicitly documented. MDCBasicColorScheme *tries* to calculate new colors, but fails in many common cases.  For example, generating an MDCBasicColorScheme with only UIColor.red will result in the `primaryLightColor` being `UIColor.red`.  This breaks some components, notably [MDCBottomNavigationBar](https://github.com/material-components/material-components-ios/blob/develop/components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m#L24) which will set both  `barTintColor` and `selectedItemTintColor` to `UIColor.red` as a result.

Personally, I'm not a fan of setting `secondaryColor` equal to `primaryColor`, but I understand it's a pretty firm requirement.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: Nullability wouldn't matter. If we set the properties to readonly, nonnull, then the implementations will need to ensure that all of the required set-up is done in `init`.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: Regarding `readwrite` properties, do we have a specific use case where a client will need to modify a scheme once it has been created?  It seems to me that creating a new scheme as a variant of another is pretty easy.

```objectivec
  MDCSemanticColorScheme *colorScheme =
      [[MDCSemanticColorScheme alloc] initWithPrimaryColor:oldScheme.primaryColor
                                  primaryColorLightVariant:oldScheme.primaryColorLightVariant
                                   primaryColorDarkVariant:oldScheme.primaryColorDarkVariant
                                            secondaryColor:UIColor.magenta
                                                errorColor:[UIColor.red colorWithAlpha:0.5]];
```

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: As far as I know, I don't think any of these are intended to be background colors.  Internally we are discussing about how far the "variants" should *vary* from the primary color, but it doesn't seem sufficient to support 4.5:1 contrast ratios.  I doubt the intention is for `.dark` to be background and `.light` to be text color (or *vice versa*)

File: components/Themes/src/MDCSemanticColorScheme.m
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.m]: In light of #2931 (UIColor and NSCoding), I intentionally went with floats here so that components using this color value won't have to do gnarly work-arounds to check color equality.  An alternative (thanks @ajsecord), might be to place a category on UIColor:

```objectivec
+ (BOOL)mdc_isApproximatelyEqualToColor:(UIColor *)otherColor;
```

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: > Should the "secondary falls back to primary" be part of the protocol or not? Right now it's only in the concrete implementation.

In the interest of less magic I think I also vote for #2, where the protocol has a nullable readonly secondary property and component themers must implement the fallback logic.

In other words: the current code is good, with the emphasis being that we need to ensure that all references to secondaryColor fall back to primaryColor in our themers.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: The primary use case I'm aware of is someone creating a Baseline color scheme and mutating its secondaryColor.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: Correct, the dark and light variants are not intended to be used for overlaying text.

File: components/Themes/src/MDCSemanticColorScheme.m
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.m]: The underlying data type for UIColor is CGFloat though, so anyone consuming a UIColor wouldn't be able to benefit from any float casting here (they can't know that we did and will have to assume the type is CGFloat).

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: We'll soon be adding onPrimary and onSecondary to the Color Scheme, which are accessibility-safe colors meant to be used for text and iconography overlaid on top of the primary or secondary color, respectively.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: I'm not sure what that would mean - do you imagine that the secondary color of the scheme would change midway during the lifecycle of an app of in the middle of user's workflow?

Are you thinking of scenario where it's not possible to create a new ColorScheme object and pass it along to all of the required consumers?  That the original has to be retained and then safely mutated?  If so, we should consider making the properties atomic.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.h]: Why not make the color scheme implement the fallback?  That way it only needs to be done once and the themers can depend on `.secondaryColor` existing, even if it's identical to `.primaryColor`.

File: components/Themes/src/MDCSemanticColorScheme.m
Comment (romoore) [reply components/Themes/src/MDCSemanticColorScheme.m]: Can you clarify what you mean, @jverkoey ? As far as I understand from Apple's docs, color data is represented internally as a sequence of either 8-bit or 16-bit values (depending on RGB or P3 color spaces).  As far as I can determine, the CGFloat values are only used for equality testing.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: Spoke offline with @romoore and we agreed on the following:

1. All of the protocol's properties should be readonly nonnull.
2. The concrete class implementation will not have any defaults or fallback logic - it will simply be a holder of the colors generated by the color tool. The color tool will provide defaults for all properties.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: Spoke offline with @romoore and agreed that the property setter case may not be particularly important and that it may be sufficient to just have a designated initializer in the first pass of the API.

File: components/Themes/src/MDCSemanticColorScheme.m
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.m]: We'll likely remove the default from the code, so this discussion can be punted til the next time we implement hex colors in code. At that point, the decision will either be:

1. Document why a (float) cast is being used instead of a CGFloat.
2. Cast to CGFloat.

**Review by jverkoey — DISMISSED** [review]

**Review by ianegordon — APPROVED** [review]
I think we can safely mark my questions as "Out of Scope"
LGTM!
Please merge once CI is happy.

**Comment by romoore** [comment]
Failing tests: 

```
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault]' started.
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:300: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault] : XCTAssertEqual failed: ("82.0") is not equal to ("70.0") - 
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:304: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault] : XCTAssertTrue failed - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault]' failed (0.022 seconds).
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth]' started.
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:322: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth] : XCTAssertEqual failed: ("64.0") is not equal to ("57.0") - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth]' failed (0.006 seconds).


Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing]' started.
components/TextFields/tests/unit/TextFieldTests.swift:235: error: -[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing] : XCTAssertEqual failed: ("56.0") is not equal to ("49.0") - 
components/TextFields/tests/unit/TextFieldTests.swift:239: error: -[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing] : XCTAssertEqual failed: ("80.0") is not equal to ("66.0") - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing]' failed (0.010 seconds).

//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3                 FAILED in 5.9s
```

**Comment by romoore** [comment]
Rebased to latest develop. Hopefully that fixes the text fields failures.

**Comment by jverkoey** [comment]
Note, we've now confirmed that the color scheme will include "on primary" and "on secondary". Feel free to add these in a follow-up PR if you prefer.

**Comment by romoore** [comment]
Text fields tests are the cause of the latest failures:

```
Test Case '-[components_TextFields_unit_test_swift_sources.MultilineTextFieldTests testSizing]' started.
components/TextFields/tests/unit/MultilineTextFieldTests.swift:158: error: -[components_TextFields_unit_test_swift_sources.MultilineTextFieldTests testSizing] : XCTAssertEqual failed: ("56.0") is not equal to ("49.0") - 
components/TextFields/tests/unit/MultilineTextFieldTests.swift:162: error: -[components_TextFields_unit_test_swift_sources.MultilineTextFieldTests testSizing] : XCTAssertEqual failed: ("80.0") is not equal to ("66.0") - 
Test Case '-[components_TextFields_unit_test_swift_sources.MultilineTextFieldTests testSizing]' failed (0.011 seconds).
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault]' started.
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:300: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault] : XCTAssertEqual failed: ("82.0") is not equal to ("70.0") - 
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:304: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault] : XCTAssertTrue failed - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyDefault]' failed (0.022 seconds).
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth]' started.
components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift:322: error: -[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth] : XCTAssertEqual failed: ("64.0") is not equal to ("57.0") - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldControllerDefaultLegacyTests testPresentationLegacyFullWidth]' failed (0.008 seconds).
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing]' started.
components/TextFields/tests/unit/TextFieldTests.swift:235: error: -[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing] : XCTAssertEqual failed: ("56.0") is not equal to ("49.0") - 
components/TextFields/tests/unit/TextFieldTests.swift:239: error: -[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing] : XCTAssertEqual failed: ("80.0") is not equal to ("66.0") - 
Test Case '-[components_TextFields_unit_test_swift_sources.TextFieldTests testSizing]' failed (0.003 seconds).

//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3                 FAILED in 5.1s
```

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (ianegordon) [reply components/Themes/src/MDCSemanticColorScheme.h]: My point is that LightVariant / DarkVariant aren't great from a semantic standpoint.
Simplifying to primaryColorVariant is more useful.  (Without light / dark)
statusBar.background = primaryColorVariant // Works in light or dark mode

**Review by ianegordon — DISMISSED** [review]

**Comment by jverkoey** [comment]
Agreed with Ian's sentiment on the light/dark variant naming. Let's move the discussion to the color design doc and see if we can flatten the color scheme's semantics so that there's just a single primaryVariant.

**Comment by jverkoey** [comment]
For now let's drop the light/dark variants from the API because they're not technically mapped to anything in iOS at this point.

**Comment by jverkoey** [comment]
Remaining action items on this PR:

- [ ] Remove the primaryLightVariant and primaryDarkVariant APIs.
- [ ] Add a primaryVariant API.
- [ ] Add an onBackgroundColor API.
- [ ] Add a surfaceColor API.
- [ ] Add an onSurfaceColor API.

**Comment by jverkoey** [comment]
Additional work:

- [ ] Expose a single init method for the Scheme type with no arguments.
- [ ] Initialize all of the parameters with the Material Defaults (documented [here](https://www.pivotaltracker.com/story/show/156171931)).
- [ ] Expose all of the Scheming properties as readwrites on the Scheme class.

We're still waiting for a team decision on whether we should expose a `+materialDefaults` API, so defer on adding that for now.

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/Themes/src/MDCSemanticColorScheme.h]: `MDCColorScheming`

File: components/Themes/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/Themes/src/MDCSemanticColorScheme.h]: We'll be dropping the light/dark variants in favor of a single variant.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:24]: I suppose we don't technically need to implement NSObject here because all of the methods are required (no need for `respondsToSelector`), but can you confirm that the conformance was intentionally left out?

Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:24]: I'm not sure, I tend not to force object requirements in protocols but I can add it. I have no feelings either way.

Comment (jverkoey) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:24]: Likely fine to leave as-is then until/if it becomes a problem :)

**Review by jverkoey — APPROVED** [review]

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:24]: Will clients need conformsToProtocol: if they are extending the protocol?

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/schemes/Color/src/MDCSemanticColorScheme.h]: super-nit / naming is hard.
Can we assume all of our schemes will be semantic and call this MDCColorScheme?

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:39]: secondaryColorVariant is listed in go/m2-color-overview.
Not here?

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/schemes/Color/src/MDCSemanticColorScheme.h]: "Material defaults"

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:79]: onError?
go/m2-color-overview

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:139]: Do we expect anyone to use this over init + properties?
What I would expect in the future is something like
- (instancetype)initWithJSONScheme: or the like.

If we keep it, please remember to add any add properties from above.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (ianegordon) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:143]: Why isn't this part of the standard declaration?
People (who are looking) will miss this and be frustrated.

**Review by ianegordon — CHANGES_REQUESTED** [review]
Minor nits.
Very excited!

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h]: Done

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h]: We cannot, since that's already a protocol name.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:39]: It wasn't in pivotal, I leave the doc to the authors.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:79]: Again, not in pivotal, so I'll let the doc authors hash-out what's going to be built.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:143]: It's more Swifty, nobody complained during the last round of reviews so I kept it. If it's a big problem, I can move it.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:39]: We'll add these parameters in a follow-up pass once the parameter mappings are more clearly defined.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:79]: We'll add these parameters in a follow-up pass once the parameter mappings are more clearly defined.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:24]: If they are using conformsToProtocol: then I think they'll have already required a type that conforms to the NSObject protocol. Within our code and our themers I don't think we'll need to do any conformity or responder testing.

Comment (ianegordon) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:24]: SGTM.
We can add it if we need it.  Taking it away after we've added it, so much harder.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (jverkoey) [inline components/schemes/Color/src/MDCSemanticColorScheme.h:139]: In the spirit of less APIs to maintain for now, let's remove this designated initializer for now and encourage use of the default initializer + property overrides.

**Review by ianegordon — APPROVED** [review]

**Comment by jverkoey** [comment]
This PR has moved to https://github.com/material-components/material-components-ios/pull/3216 so that we can run CI against develop and land.

File: components/schemes/Color/src/MDCSemanticColorScheme.h
Comment (romoore) [reply components/schemes/Color/src/MDCSemanticColorScheme.h:139]: I thought this was being dropped?

**Pull request closed without merging**
